Update README.md (#3149)
medplum/medplum: README.md

![Medplum](packages/docs/static/img/cover.webp)

> Medplum is a comprehensive developer platform dedicated to simplifying and enhancing healthcare application development. With a strong focus on security, compliance, and interoperability, Medplum offers a range of interconnected components and tools designed to streamline the creation of healthcare applications and systems.

Reviewer: I mostly like this change, but think it could be a bit tighter: it reads as pretty verbose to me in its current form.  Maybe something like this could work:
```suggestion
Medplum is a comprehensive developer platform that simplifies and enhances healthcare application development. With a strong focus on security, compliance, and simple integrations, Medplum provides the foundational components and tools needed to build the healthcare applications of the future.
```

#### Technologies

Let's discuss the key technologies that power this monorepo, providing a foundation for our healthcare application development:

> - **Full-stack TypeScript**: Our codebase is primarily built using TypeScript, enabling type-safe development across both the server and frontend components. Node.js is used for server-side code, while React powers the frontend applications.

Reviewer: This change reads a lot better to me, and does a good job explaining more of the rationale for TypeScript

Author: I agree, cheers! _

> Here is a list of the major technologies in use:

Reviewer: Maybe this line should be removed, and the lists combined?

- **PostgreSQL**: Our choice for data storage, ensuring a robust and reliable foundation for healthcare data.

> - **Redis**: Used for background jobs and caching, enhancing the performance and responsiveness of our applications.

Reviewer: ```suggestion
- **Redis**: Used for background jobs and caching, enhancing the performance and responsiveness of the server.
```

> - **Express**: Serving as our API server, Express simplifies routing and handling requests, facilitating efficient data exchange.

Reviewer: "Serving as [a] server" sounds a bit tautological to me — consider something like this instead:
```suggestion
- **Express**: Serving as our API framework, Express simplifies routing and handling API requests
```

> - **TypeScript**: A type-safe extension of JavaScript, TypeScript enhances code quality and maintainability, crucial for healthcare application development.

Reviewer: This repeats some verbiage from above, which could simply be removed
```suggestion
- **TypeScript**: A type-safe extension of JavaScript, TypeScript enhances code quality and maintainability
```

> - **React**: As the driving force behind our frontend applications, React offers a versatile and component-based approach to creating intuitive user interfaces.

Reviewer: ```suggestion
- **React**: React offers a versatile, component-based approach to creating web UIs
```
